### PR TITLE
Bump version to 0.19

### PR DESCRIPTION
**Pull request by Wauplin**
Follow-up after 0.18 release.

`InferenceAPI` was announced to be removed by now but I don't think it's wise. Given how much it has being used in external libraries and scripts I think it's wiser to deprecate it with no real deadline (I've set `v1.0` but not sure it'll happen someday^^).

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
LGTM!

**Comment by jamesbraza** [comment]
Would you be willing to hold off on this release for one day while I add support for pydantic v2? 😄

**Comment by Wauplin** [comment]
@jamesbraza Actually the `0.18.0` is already out since the beginning of the week: https://github.com/huggingface/huggingface_hub/releases/tag/v0.18.0.

This PR just sets the version to `0.19.0.dev0` until the `0.19` is properly released. But that will only be in a few weeks when we have enough new features to release. In any case you can always install `huggingface_hub` from the source so once pydantic v2 is merged, you'll be able to do
```
pip install git+https://github.com/huggingface/huggingface_hub@main
```

**Comment by jamesbraza** [comment]
Ah I see, thanks!

**Pull request merged**
